### PR TITLE
Graticule labels

### PR DESCRIPTION
**Pull request by ahocevar**
This pull request replaces #4369. It provides the same API. I implemented things quite differently to improve performance, and tried to use sensible defaults so graticule labels work fine out of the box.

### CODE REVIEW

File: src/ol/graticule.js [hunk 1]
```diff
@@ -104,31 +108,118 @@ ol.Graticule = function(opt_options) {
...
+  this.parallelsLabels_ = null;
+
+  if (options.showLabels == true) {
+    var degreesToString = ol.coordinate.degreesToStringHDMS;
+
+    /**
+     * @type {null|function(number):string}
+     * @private
+     */
+    this.lonLabelFormatter_ = options.latLabelFormatter == undefined ?
```
Comment (bartvde) [inline src/ol/graticule.js]: should be ```options.lonLabelFormatter``` instead

File: src/ol/graticule.js [hunk 2]
```diff
@@ -104,31 +108,118 @@ ol.Graticule = function(opt_options) {
...
+        degreesToString.bind(this, 'EW') :
+        options.latLabelFormatter;
```
Comment (bartvde) [inline src/ol/graticule.js]: same here

**Comment by fredj** [comment]
nice improvement, thanks!

File: src/ol/graticule.js (hunk 2 above)
Comment (ahocevar) [inline src/ol/graticule.js]: Thanks for the review. I fixed this and also modified the tests to catch it.

Comment (bartvde) [inline src/ol/graticule.js]: Great, I was thinking the same when reading the tests since they had the same function bodies for the formatter functions

**Comment by cmdelatorre** [comment]
Hello, I have a question: is this feature available in OL 4.1.1 ?

**Comment by ahocevar** [comment]
No, but it will be in v4.2.0, which will be released tomorrow.

**Comment by cmdelatorre** [comment]
Awesome! Great news! Thank you very much for the great work!

**Pull request merged**